### PR TITLE
Points and levels

### PR DESCRIPTION
**Pull request by lramirez**
* User has points, not level.
* Level determined by UserService.
* Each level requires 1000 more points.
* User gets points when a task is done.

### CODE REVIEW

File: src/main/java/pingis/entities/TaskInstance.java [hunk 1]
```diff
@@ -143,5 +143,4 @@ public Challenge getChallenge() {
     return this.task.getChallenge();
   }
 
-
```
Comment (dwarfcrank) [inline src/main/java/pingis/entities/TaskInstance.java]: Unnecessary whitespace change.

File: src/main/java/pingis/entities/User.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
   @NotNull
   @Min(LEVEL_MIN_VALUE)
```
Comment (dwarfcrank) [inline src/main/java/pingis/entities/User.java]: These constraints should be either removed or updated.

File: src/main/java/pingis/services/UserService.java [hunk 3]
```diff
@@ -101,4 +105,17 @@ public User findByName(String name) {
   public User getCurrentUser() {
     return findByName(SecurityContextHolder.getContext().getAuthentication().getName());
   }
+
+  public int getLevel(int score) {
+    for (int i = 1; i <= numberOfLevels; i++) {
```
Comment (dwarfcrank) [inline src/main/java/pingis/services/UserService.java]: Couldn't this be computed with logarithms instead, e.g. `level = log(points, base) - log(pointsForFirstLevel, base)`?

Comment (dwarfcrank) [reply src/main/java/pingis/services/UserService.java]: Also needs more tests.

File: src/test/java/pingis/entities/UserTest.java [hunk 4]
```diff
@@ -28,9 +28,9 @@ public void testUserName() {
   }
 
   @Test
-  public void testUserLevel() {
+  public void testUserPoints() {
     final double level = 0.001;
```
Comment (dwarfcrank) [inline src/test/java/pingis/entities/UserTest.java]: Why is this named `level` when it's used as the minimum delta for float-point comparisons?

**Comment by lramirez** [comment]
Accidentally included commits from another branch.

File: src/main/java/pingis/services/UserService.java [hunk 5]
```diff
@@ -13,6 +13,10 @@
 @Service
 public class UserService {
 
+  public static final int numberOfLevels = 100;
+  private final int pointsForFirstLevel = 1000;
+  private final float base = 1.2f;
```
Comment (Heliozoa) [inline src/main/java/pingis/services/UserService.java]: Unused variables?

File: src/test/java/pingis/entities/UserTest.java (hunk 4 above)
Comment (Heliozoa) [inline src/test/java/pingis/entities/UserTest.java]: Echoing @dwarfcrank

File: src/main/java/pingis/services/DataImporter.java [hunk 6]
```diff
@@ -29,9 +29,9 @@
 @Component
 public class DataImporter {
 
-  private static final int TMC_USER_LEVEL = 100;
-  private static final int TEST_USER_LEVEL = 5;
-  private static final int IMPLEMENTATION_USER_LEVEL = 1;
+  private static final int TMC_USER_LEVEL = 100000;
+  private static final int TEST_USER_LEVEL = 5000;
+  private static final int IMPLEMENTATION_USER_LEVEL = 1000;
```
Comment (Heliozoa) [inline src/main/java/pingis/services/DataImporter.java]: Unused variables? Also, the name should probably be changed to points.

File: src/main/java/pingis/entities/TaskInstance.java [hunk 7]
```diff
@@ -139,8 +139,9 @@ public void addImplementationTaskInstance(TaskInstance implementationTaskInstanc
   public Challenge getChallenge() {
     return this.task.getChallenge();
   }
-
+
```
Comment (Heliozoa) [inline src/main/java/pingis/entities/TaskInstance.java:142]: Whitespace change

File: src/main/java/pingis/entities/Task.java [hunk 8]
```diff
@@ -22,7 +22,7 @@
   public static final int NAME_MIN_LENGTH = 3;
   public static final int NAME_MAX_LENGTH = 50;
   public static final int LEVEL_MIN_VALUE = 1;
-  public static final int LEVEL_MAX_VALUE = 200;
+  public static final int LEVEL_MAX_VALUE = 200000;
```
Comment (Heliozoa) [inline src/main/java/pingis/entities/Task.java]: Should probably be renamed to points

**Review by Heliozoa — CHANGES_REQUESTED** [review]

File: src/test/java/pingis/services/TaskInstanceServiceTest.java [hunk 9]
```diff
@@ -163,8 +163,10 @@ public void testUpdateTaskInstanceCode() {
 
   @Test
   public void testMarkAsDone() {
+    int oldPoints = testUser.getPoints();
     TaskInstance result = taskInstanceService.markAsDone(testTaskInstance);
     assertEquals(CodeStatus.DONE, result.getStatus());
+    assertTrue(oldPoints < testUser.getPoints());
```
Comment (dwarfcrank) [inline src/test/java/pingis/services/TaskInstanceServiceTest.java]: This test would be better if you calculated the expected number of points and used `assertEquals`.

**Review by Heliozoa — APPROVED** [review]

**Pull request merged**